src/bin/extract_metadata.c

```diff
@@ -69,21 +69,41 @@ static int handle_variable_sav(int index, readstat_variable_t *variable, const c
     const char *label = readstat_variable_get_label(variable);
     int decimals = -1;
 
-    if (readstat_variable_get_type_class(variable) == READSTAT_TYPE_CLASS_STRING) {
+    switch (readstat_variable_get_type_class(variable)) {
+    case READSTAT_TYPE_CLASS_STRING:
         type = "STRING";
-    } else if (readstat_variable_get_type_class(variable) == READSTAT_TYPE_CLASS_NUMERIC) {
-        if (format && (strncmp(format, "DATE", sizeof("DATE")-1) == 0 || 
-                    strncmp(format, "ADATE", sizeof("ADATE")-1) == 0 || 
-                    strncmp(format, "EDATE", sizeof("EDATE")-1) == 0 || 
+        break;
+    case READSTAT_TYPE_CLASS_NUMERIC:
+        // Check for date and duration types
+        // https://libguides.library.kent.edu/SPSS/DatesTime
+        if (format && (strncmp(format, "DATE", sizeof("DATE")-1) == 0 ||
+                    strncmp(format, "ADATE", sizeof("ADATE")-1) == 0 ||
+                    strncmp(format, "EDATE", sizeof("EDATE")-1) == 0 ||
                     strncmp(format, "SDATE", sizeof("SDATE")-1) == 0)) {
```

Comment:
Since `format` is a regular C string, `strcmp` instead of `strncmp` will be fine here.